src/coreclr/inc/corinfo.h

```diff
@@ -2441,6 +2441,7 @@ class ICorStaticInfo
 
     virtual void getReadyToRunDelegateCtorHelper(
             CORINFO_RESOLVED_TOKEN * pTargetMethod,
+            mdToken                  targetConstraint,
```

Comment:
Would it make sense to pass the full resolved token, like for `getCallInfo`?

Reply:
The info in `pTargetMethod` would be redundant with the new information passed in `targetConstraint` (the scope/context of the token is the same, and we need to resolve the token on the EE side anyway because the `__Canon` that we give to RyuJIT is not precise enough for EE purposes). It felt like a waste of memory to keep track of the extra state but also this is going to be rare enough that the extra memory probably doesn't matter...

I'm indifferent, slightly leaning toward not touching SPMI again for this pull request.

Reply:
Sounds fine to me. I mainly asked because I saw the other pattern in `getCallInfo` and am not super familiar with how this metadata is passed back and forth.

Reply:
The general pattern used by JIT/EE interface is to either pass around resolve handle (e.g. `CORINFO_CLASS_HANDLE`) or the full `CORINFO_RESOLVED_TOKEN` structure when additional details are needed. The motivation for this pattern is to avoid resolving same token multiple times.

In this particular case, I think it would be most appropriate to pass `CORINFO_CLASS_HANDLE` for the constraint token.

Reply:
`CORINFO_CLASS_HANDLE` doesn't have the necessary information - for cases like:

```csharp
void Something<T, U> where T : IInterface, U: IInterface
{
    Action a = T.InterfaceMethod;
    Action b = U.InterfaceMethod;
}
```

When compiling `Something<__Canon,__Canon>`, the `CORINFO_CLASS_HANDLE` in both cases is `__Canon` because RyuJIT doesn't care about the distinction, but we need to be able to disambiguate between those two cases.

We need the token so that we can grab the type in a runtime-determined form. It's also why the resolved form of the token stored in `CORINFO_RESOLVED_TOKEN` is not that useful.

Reply:
Got it